src/react/components/pages/prebuiltPredict/prebuiltPredictPage.tsx

```diff
@@ -923,45 +924,53 @@ export class PrebuiltPredictPage extends React.Component<IPrebuiltPredictPagePro
         this.setState({ predictionEndpointUrl: this.getUpdateRequestURI() });
     }
 
-    private getUpdateRequestURI = () => {
-        const { prebuiltSettings } = this.props;
-        const { currentPrebuiltType, predictionEndpointUrl } = this.state;
-        let updatedURI = "";
-        if (predictionEndpointUrl.includes("?")) {
-            const queryString = predictionEndpointUrl.split("?")[1];
-            if (prebuiltSettings && prebuiltSettings.serviceURI) {
-                const parameterArray = queryString.includes("&") ? queryString.split("&") : [queryString];
-                let newQueryString = "";
-                let connector = "";
-                for (const parameter of parameterArray) {
-                    const name = parameter.split("=")[0];
-                    if (name !== "locale" && name !== constants.pages) {
-                        newQueryString += `${connector}${parameter}`;
-                    }
-                    connector = "&";
-                }
-                if (this.state.withPageRange && this.state.pageRangeIsValid) {
-                    newQueryString += `${connector}${constants.pages}=${this.state.pageRange}`;
-                    connector = "&";
+    private getUpdateRequestURI = (fromTextArea = false) => {
+        const pathTemp = "/formrecognizer/:prebuiltServiceVersion/prebuilt/:prebuiltType/analyze";
+        const { predictionEndpointUrl } = this.state;
+        const [path, queryString] = predictionEndpointUrl.split("?");
+        const pathParams = URIUtils.matchPath(pathTemp, URIUtils.normalizePath(path));
+        if (fromTextArea) {
+            const prebuiltType = _.get(pathParams, "prebuiltType", "")
+            if (prebuiltType && prebuiltType !== this.state.currentPrebuiltType.servicePath) {
+                const ret = this.prebuiltTypes.filter(item => item.servicePath === prebuiltType)
+                if (ret.length === 1) {
+                    this.setState({ currentPrebuiltType: ret[0] })
                 }
-                if (this.state.currentPrebuiltType.useLocale) {
-                    newQueryString += `${connector}locale=${this.state.currentLocale}`;
+            }
+        } else {
+            pathParams["prebuiltType"] = this.state.currentPrebuiltType.servicePath;
+        }
+        const newPath = URIUtils.compilePath(pathTemp, pathParams);
```

Comment:
nit:
1. Add a blank line below.
2. This function actually handle path and querystring params separately, ideally we can have two new functions helping on those part respectivly?